pkg/scaling/scale_handler.go

```diff
@@ -441,14 +451,30 @@ func (h *scaleHandler) GetScaledObjectMetrics(ctx context.Context, scaledObjectN
 			cache.Recorder.Event(scaledObject, corev1.EventTypeWarning, eventreason.KEDAScalerFailed, err.Error())
 		}
 
+		if len(metricsArray) == 0 {
+			err = fmt.Errorf("no metrics found getting metricsArray array %s", metricsName)
+			logger.Error(err, "error metricsArray is empty")
+			// TODO: add cache.Recorder?
```

Comment:
what about this TODO?

Reply:
Im not sure if the cache.Recorder should be added there as well. If not i can just delete it. wdyt?